interfaces/connection.go

```diff
@@ -49,29 +50,51 @@ type ConnectedSlot struct {
 // Attrer is an interface with Attr getter method common
 // to ConnectedSlot, ConnectedPlug, PlugInfo and SlotInfo types.
 type Attrer interface {
-	Attr(key string, val interface{}) error
+	// Attr returns attribute value for given path, or an error. Dotted paths are supported.
+	Attr(path string, value interface{}) error
+	// Lookup returns attribute value for given path, or false. Dotted paths are supported.
+	Lookup(path string) (value interface{}, ok bool)
 }
 
-func getAttribute(snapName string, ifaceName string, staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, key string, val interface{}) error {
+func lookupAttr(staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, path string) (interface{}, bool) {
 	var v interface{}
-	var ok bool
+	comps := strings.FieldsFunc(path, func(r rune) bool { return r == '.' })
+	if len(comps) == 0 {
```

Comment:
If I pass `foo` as path why would I get `nil, false` as response?

Reply:
You won't, FieldsFunc will return ["foo"].

Reply:
Added one more test case to highlight this (although it's already tested indirectly via tests of `.Attr` methods).